haskell-src/exec/Chainweb/Server.hs

```diff
@@ -458,6 +481,25 @@ queryTxsByKey logger rk c =
        return ev
     return $ (`fmap` r) $ \(tx,contHist, blk) -> toApiTxDetail tx contHist blk evs
 
+queryTxsByPactId :: LogFunctionIO Text -> Text -> Connection -> IO [TxSummary]
+queryTxsByPactId logger pactid c =
+  runBeamPostgresDebug (logger Debug . T.pack) c $ do
+    r <- runSelectReturningList (queryTxsByPactId' pactid)
+    return $ (`fmap` r) $ \(tx,contHist, blk) -> toApiTxSummary tx contHist blk
+
+queryTxsByPactId' :: Text -> SqlSelect Postgres (TransactionT Identity, ContinuationHistoryT Identity, BlockT Identity)
+queryTxsByPactId' pactid =
+    select $ orderBy_ statusOrd $ do
+      tx <- all_ (_cddb_transactions database)
+      contHist <- joinContinuationHistory (_tx_pactId tx)
+      blk <- all_ (_cddb_blocks database)
+      guard_ (_tx_block tx `references_` blk)
```

Comment:
I actually don't think we need to join on the `block` row here.

Reply:
After removing the join with the blocks table, here is the resulting query plan.
```
                                                                               QUERY PLAN                                                         
------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Nested Loop  (cost=876.78..30818.30 rows=35 width=1963) (actual time=0.124..0.126 rows=0 loops=1)
   ->  Index Scan using transactions_pactid_index on transactions t0  (cost=0.42..144.91 rows=35 width=1899) (actual time=0.124..0.124 rows=0 loops=1)
         Index Cond: ((pactid)::text = 'apactid'::text)
   ->  Aggregate  (cost=876.35..876.36 rows=1 width=64) (never executed)
         CTE transactionsteps
           ->  Recursive Union  (cost=0.56..873.82 rows=101 width=282) (never executed)
                 ->  Unique  (cost=0.56..8.57 rows=1 width=282) (never executed)
                       ->  Index Scan using transactions_requestkey_idx on transactions tinner  (cost=0.56..8.57 rows=1 width=282) (never executed)
                             Index Cond: ((requestkey)::text = ($1)::text)
                 ->  Subquery Scan on "*SELECT* 2"  (cost=86.27..86.42 rows=10 width=282) (never executed)
                       ->  Unique  (cost=86.27..86.32 rows=10 width=286) (never executed)
                             ->  Sort  (cost=86.27..86.30 rows=11 width=286) (never executed)
                                   Sort Key: trec.depth
                                   ->  Nested Loop  (cost=0.56..86.08 rows=11 width=286) (never executed)
                                         ->  WorkTable Scan on transactionsteps trec  (cost=0.00..0.20 rows=10 width=36) (never executed)
                                         ->  Index Scan using transactions_requestkey_idx on transactions tinner_1  (cost=0.56..8.57 rows=1 width=278) (never executed)
                                               Index Cond: ((requestkey)::text = (trec.pactid)::text)
         ->  CTE Scan on transactionsteps  (cost=0.00..2.02 rows=101 width=68) (never executed)
 Planning Time: 1.210 ms
 Execution Time: 0.287 ms
```